<launch>

  <arg name="gui" default="true" />

  <include file="$(find recognition_3d)/launch/publish_dataset.launch">
    <arg name="gui" value="false" />
  </include>

  <node name="object_array_publisher"
        pkg="jsk_recognition_msgs" type="object_array_publisher.py"
        output="screen">
    <remap from="~output" to="/object_array" />
    <rosparam subst_value="true">
      latch: true
      objects:
        <!-- - id: 1 -->
        <!--   name: avery_binder -->
        <!--   image_resources: -->
        <!--     - package://jsk_arc2017_common/data/objects/avery_binder/top.jpg -->
        <!--   mesh_resource: package://jsk_arc2017_common/data/objects/avery_binder/mesh/mesh.obj -->
        - id: 1
          name: pet_bottle
          image_resources:
            - file://$(optenv HOME)/dataset/mesh_dataset_shapenet/pet_bottle_1/images/texture0.jpg
          mesh_resource: file://$(optenv HOME)/dataset/mesh_dataset_shapenet/pet_bottle_1/models/model_normalized.obj
    </rosparam>
  </node>

  <node name="transformable_interactive_server"
        pkg="jsk_interactive_marker" type="transformable_server_sample">
    <rosparam subst_value="true">
      display_interactive_manipulator: true
      display_interactive_manipulator_only_selected: true
      interaction_mode: 2
    </rosparam>
  </node>

  <include file="$(find recognition_3d)/launch/save_object_pose.launch">
    <arg name="OBJECT_ID" value="f49d6c4b75f695c44d34bdc365023cf4"/>
    <arg name="OBJECT_FRAME" value="pet_bottle"/>
  </include>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find recognition_3d)/rvizconfig/object_3d_annotation.rviz">
    </node>
    <node name="rqt_reconfigure"
          pkg="rqt_reconfigure" type="rqt_reconfigure"
          args="publish_dataset">
    </node>
  </group>

</launch>
